test/source/tests/runtime_tests.cpp

```diff
@@ -58,6 +71,30 @@ void concurrencpp::tests::test_runtime_constructor() {
     assert_equal(runtime.thread_pool_executor()->max_worker_idle_time(), opts.max_thread_pool_executor_waiting_time);
     assert_equal(runtime.background_executor()->max_concurrency_level(), opts.max_background_threads);
     assert_equal(runtime.background_executor()->max_worker_idle_time(), opts.max_background_executor_waiting_time);
+
+    auto test_runtime_executor = [&thread_started_callback_invocations_num,
+                                  &thread_terminated_callback_invocations_num](std::shared_ptr<executor> executor) {
+        size_t start_thread_started_callback_invocations_num = thread_started_callback_invocations_num;
```

Comment:
why do we actually do it instead of just setting  `thread_started_callback_invocations_num`/ `thread_terminated_callback_invocations_num` to `0` in the start of the function? then we can just test against `0`, `1`